multiqc/modules/biscuit/biscuit.py

```diff
@@ -0,0 +1,688 @@
+            for cov in covs:
+                ccov_cnts.append(_ccov_cnt/1000000.)
+                _ccov_cnt -= dd[cov]
+
+            data = dict(zip(covs,ccov_cnts))
+        else:
+            raise Exception("Unknown file received: %s" % fn)
+                
+        return data
+
+    def chart_biscuit_coverage(self):
+
+        # base coverage
+        basecov = OrderedDict()
+        mdata = [
+            dict([(k.replace('_covdist_table.txt',''),v) for k, v in self.mdata['coverage'].items() if k.endswith('_covdist_table.txt')]),
+            dict([(k.replace('_covdist_q40_table.txt',''),v) for k, v in self.mdata['coverage'].items() if k.endswith('_covdist_q40_table.txt')]),
+            dict([(k.replace('_covdist_q40_botgc_table.txt',''),v) for k, v in self.mdata['coverage'].items() if k.endswith('_covdist_q40_botgc_table.txt')]),
+            dict([(k.replace('_covdist_q40_topgc_table.txt',''),v) for k, v in self.mdata['coverage'].items() if k.endswith('_covdist_q40_topgc_table.txt')]),
+            dict([(k.replace('_covdist_cpg_table.txt',''),v) for k, v in self.mdata['coverage'].items() if k.endswith('_covdist_cpg_table.txt')]),
+            dict([(k.replace('_covdist_cpg_q40_table.txt',''),v) for k, v in self.mdata['coverage'].items() if k.endswith('_covdist_cpg_q40_table.txt')]),
+            dict([(k.replace('_covdist_cpg_q40_botgc_table.txt',''),v) for k, v in self.mdata['coverage'].items() if k.endswith('_covdist_cpg_q40_botgc_table.txt')]),
+            dict([(k.replace('_covdist_cpg_q40_topgc_table.txt',''),v) for k, v in self.mdata['coverage'].items() if k.endswith('_covdist_cpg_q40_topgc_table.txt')]),
+        ]
+        if len(mdata)>0:
+            self.add_section(
+                name = 'Cumulative Base Coverage',
+                anchor = 'biscuit-coverage-base',
+                description = "<p>This plot shows the cummulative base coverage. High and low GC content region are the top and bottom 10% 100bp window in GC content.</p>",
+                plot = linegraph.plot(mdata, {'id':'biscuit_coverage_base',
+                    'title': 'Cumulative Base Coverage',
+                    'data_labels': [
+                        {'name': 'All', 'ylab':'Million Bases'}, 
+                        {'name': 'Q40', 'ylab':'Million Bases'}, 
+                        {'name': 'Q40 low GC', 'ylab':'Million Bases'}, 
+                        {'name': 'Q40 high GC', 'ylab':'Million Bases'}, 
+                        {'name': 'CpG (all)', 'ylab':'Million CpGs'}, 
+                        {'name': 'CpG Q40', 'ylab':'Million CpGs'}, 
+                        {'name': 'CpG Q40 low GC', 'ylab':'Million CpGs'}, 
+                        {'name': 'CpG Q40 high GC', 'ylab':'Million CpGs'}, 
+                    ]})
+            )
+            
+            for sid, dd in mdata[0].items():
+                if sid not in basecov:
+                    basecov[sid] = {}
+                basecov[sid]['all'] = dd[1]/dd[0]*100
```

Comment:
Note that you want to be careful with ints and floats here. `100` is an `int`, so if the others are parsed as ints then the result will also be an integer. I usually wrap stuff in `float()` and use `100.0` to force floating number results.

Reply:
Thanks for the warning. I added `float()` and `100.0` to force floating number calculation.